modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -424,6 +432,27 @@ class Session implements ISession {
         return this
     }
 
+    /**
+     * Downloads the custom fsync file from the specified URL.
+     */
+    void downloadCustomFsync() {
+        Path customFsyncPath = workDir.resolve("custom_fsync")
+        if (customFsyncPath.exists()) {
+            log.debug "Custom fsync was already downloaded"
+            return
+        }
+
+        try {
+            URLConnection conn = new URL(CUSTOM_FSYNC_URL).openConnection()
+            conn.connect()
+            Files.copy(conn.inputStream, customFsyncPath)
+            customFsyncPath.setExecutable(true, false)
+            log.info "File downloaded successfully: $customFsyncPath"
+        } catch (IOException e) {
+            log.error "Failed to download file: $CUSTOM_FSYNC_URL", e
```

Comment:
will this command fail for every task if the download fails? 
i think if the download fails we should diable the execution of the `custom_fsync` command or maybe abort the execution? 
if we add retries it shouldn't fail frequently right?

Reply:
The tasks won't fail if the download fails, it will just log the error. If the tasks can't find the binary, they will just print the error to the log, but won't fail